libs/execution_context/ebpf_core.c

```diff
@@ -728,6 +728,132 @@ _ebpf_core_protocol_get_program_information(
+    _In_ uint16_t entry_count,
+    _In_count_(entry_count) ebpf_pinning_entry_t* pinning_entries,
+    _Outptr_result_buffer_maybenull_(entry_count) ebpf_core_map_information_t** map_info)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    ebpf_core_map_information_t* local_map_info = NULL;
+    uint16_t index;
+
+    if (map_info == NULL) {
+        result = EBPF_INVALID_ARGUMENT;
+        goto Exit;
+    }
+
+    if ((entry_count == 0) || (pinning_entries == NULL))
+        goto Exit;
+
+    local_map_info = (ebpf_core_map_information_t*)ebpf_allocate(
+        sizeof(ebpf_core_map_information_t) * entry_count, EBPF_MEMORY_NO_EXECUTE);
+    if (local_map_info == NULL) {
+        result = EBPF_NO_MEMORY;
+        goto Exit;
+    }
+
+    for (index = 0; index < entry_count; index++) {
+        ebpf_pinning_entry_t* source = &pinning_entries[index];
+        ebpf_core_map_information_t* destination = &local_map_info[index];
+        ebpf_map_definition_t* map_definition;
+        if (EBPF_OBJECT_MAP != ebpf_object_get_type(source->object)) {
+            // Bad object type.
+            result = EBPF_INVALID_ARGUMENT;
+            goto Exit;
+        }
+        // Query map defintion.
+        map_definition = ebpf_map_get_definition((ebpf_map_t*)source->object);
+        destination->definition = *map_definition;
+        // Set pin path. No need to duplicate.
+        destination->pin_path = source->name;
+    }
+
+Exit:
+    if (result != EBPF_SUCCESS) {
+        ebpf_free(local_map_info);
+        local_map_info = NULL;
+    }
+
+    *map_info = local_map_info;
+    return result;
+}
+
+ebpf_result_t
+_ebpf_core_protocol_serialize_map_information_reply(
+    _In_ const uint16_t map_count,
+    _In_count_(map_count) const ebpf_core_map_information_t* map_info,
+    _In_ ebpf_operation_get_map_information_reply_t* map_info_reply,
+    _In_ const size_t output_buffer_length)
+{
+    ebpf_result_t result = EBPF_SUCCESS;
+    size_t serialization_buffer_size;
+    size_t required_serialization_length;
+
+    serialization_buffer_size = output_buffer_length - EBPF_OFFSET_OF(ebpf_operation_get_map_information_reply_t, data);
+
+    result = ebpf_serialize_core_map_information_array(
+        map_count,
+        map_info,
+        map_info_reply->data,
+        (const size_t)serialization_buffer_size,
+        (size_t*)&map_info_reply->size,
+        &required_serialization_length);
+
+    if (result != EBPF_SUCCESS) {
+        map_info_reply->header.length = (uint16_t)(
+            required_serialization_length + EBPF_OFFSET_OF(ebpf_operation_get_map_information_reply_t, data));
+    } else
+        map_info_reply->map_count = map_count;
+
+    return result;
+}
+
+static ebpf_result_t
+_ebpf_core_protocol_get_map_information(
+    _In_ const ebpf_operation_get_map_information_request_t* request,
+    _In_ ebpf_operation_get_map_information_reply_t* reply,
+    _In_ uint16_t reply_length)
```

Comment:
never use SAL _In_ annotation on value types, only pointer types
```suggestion
    uint16_t reply_length)
```
Please fix everywhere else (line 784, 787, etc.)